course7/homeworks/patrickh/homework3/calculator.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stdexcept>
+#include <iostream>
+#include "calculator.h"
+
+void Calculator::parse(const std::string& input)
+{
+    if (input == "+")
+        add();
+    else if (input == "-")
+        subtract();
+    else if (input == "*")
+        multiply();
+    else if (input == "/")
+        divide();
+    else
+        mStack.emplace(input);
```

Comment:
I think it's more clear to explicitly use the Complex constructor here, instead of replying on an implicit cast.